### DISCUSSION TITLE
GUI-Based Twitch Chat Bot To VRChat OSC via Commands

### DISCUSSION THREAD

**Discussion by AcChosen** [vrchat-community/osc#101] [Show and tell]
A really simple but customizable Twitch bot with a GUI interface that takes chat commands and sends them to VRChat through OSC.
Written in Node JS with Electron.

![alt text](https://i.imgur.com/QtTP92d.png)

## Features:
- Up to 12 commands with custom addressses, values, value types, and a bot chat response.
- Supports sending Int, Float, and Bool OSC messages.
- Optional Timed feature to allow a command to fire a second OSC message after a determined amount of time.
- Optional delay fuction to prevent chat from spamming commands to much.
- Optional setting of custom incoming and outgoing ports.
- Optional custom bot accounts. Unchecking the "Use default public bot" checkbox will allow you to enter a custom Twitch account and OAUTH token for the bot to live on instead.
- Ability to save and load your commands. After entering all your information in, you can save it to a .vrcosc file and open it again for later, in case you have different avatars with different commands!
- Toggleable whitelist! Allows the bot to ignore commands from anyone who isn't on the list if the list is enabled.

Video Preview: https://twitter.com/ac_chosen/status/1499520193822920704

Repository: https://github.com/AcChosen/EZTwitchOSCBot

**Comment by momo-the-monster** [comment]
Thanks for sharing this, @AcChosen ! It's been added to the [list](https://github.com/vrchat-community/osc/wiki/Resources#twitch).
